compiler/rustc_mir_build/src/builder/expr/as_rvalue.rs

```diff
@@ -656,6 +657,24 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         block.and(rvalue)
     }
 
+    /// Recursively inspect a THIR expression and probe through unsizing
+    /// operations that can be const-folded today.
```

Comment:
What does this have to do with const-folding (which usually refers to the constant propagation optimization pass)?